R/system.R

```diff
@@ -1616,8 +1616,18 @@ parse_html_tables <- function(url, encoding = NULL) {
 scrape_html_table <- function(url, index, heading, encoding = NULL) {
   loadNamespace("rvest");
   loadNamespace("tibble");
+  current_locale = Sys.getlocale(category = "LC_CTYPE")
   .htmltables <- parse_html_tables(url, encoding)
-  tibble::repair_names(rvest::html_table(.htmltables[[index]], fill=TRUE ,header=heading))
+  if (Sys.info()["sysname"] == "Windows") {
+    # For Windows, temporary change LC_CTYPE to C to workaround the "invalid multibyte string" error raised from rvest.
+    Sys.setlocale(category="LC_CTYPE", locale="C");
+  }
+  res <- tibble::repair_names(rvest::html_table(.htmltables[[index]], fill=TRUE ,header=heading))
+  if (Sys.info()["sysname"] == "Windows") {
```

Comment:
Use tryCatch to make sure locale gets back to original even in error case.

Reply:
Changed to use tryCacth to make sure locale gets back to the original one even with the error case.